File: 240-search-a-2d-matrix-ii/240-search-a-2d-matrix-ii.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], tar: int) -> bool:
        
        import bisect as bi 
        for row in matrix: 
            ix = bi.bisect_left(row, tar)
            if ix >=len(row): continue 
            if row[ix] == tar: return True 
        return False  
            

File: 240-search-a-2d-matrix-ii/240-search-a-2d-matrix-ii.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& mat, int tar) {
        
        int R = mat.size() , C = mat[0].size() ; 
        int r = 0 , c = C-1 ; 
        while(r < R && c>=0 ) {
            if(mat[r][c] == tar) return true ; 
            mat[r][c] > tar? c--:r++ ; 
        }
        return false ; 
        
    }
};